test-data/unit/check-columns.test

```diff
@@ -65,4 +45,175 @@ def f(x: object, n: int, s: str) -> None:
             s = x # E:13: Incompatible types in assignment (expression has type "int", variable has type "str")
         n = x # E:9: Incompatible types in assignment (expression has type "object", variable has type "int")
 [builtins fixtures/isinstance.pyi]
-[out]
+
+[case testColumnHasNoAttribute]
+import m
+if int():
+    from m import foobaz # E:5: Module 'm' has no attribute 'foobaz'; maybe "foobar"?
+(1).x # E:2: "int" has no attribute "x"
+(m.foobaz()) # E:2: Module has no attribute "foobaz"
+
+[file m.py]
+def foobar(): pass
+
+[builtins fixtures/module.pyi]
+
+[case testColumnUnexpectedKeywordArg]
+def f(): pass
+(f(x=1)) # E:2: Unexpected keyword argument "x" for "f"
+
+[case testColumnDefinedHere]
+class A: pass
+if int():
+    def f(a: 'A') -> None: pass # N:5: "f" defined here
+    (f(b=object())) # E:6: Unexpected keyword argument "b" for "f"
+
+[case testColumnInvalidType]
+# TODO: Add column numbers
+from typing import Iterable
+
+bad = 0
+
+def f(x: bad): # E: Invalid type "__main__.bad"
+    y: bad # E:8: Invalid type "__main__.bad"
+
+def g(x): # E: Invalid type "__main__.bad"
+    # type: (bad) -> None
+    y = 0  # type: bad  # E: Invalid type "__main__.bad"
+
+z: Iterable[bad] # E: Invalid type "__main__.bad"
+
+[case testColumnFunctionMissingTypeAnnotation]
+# flags: --disallow-untyped-defs
+if int():
+    def f(x: int): # E:5: Function is missing a return type annotation
+        pass
+
+    def g(x): # E:5: Function is missing a type annotation
+        pass
+
+[case testColumnNameIsNotDefined]
+((x)) # E:3: Name 'x' is not defined
+
+[case testColumnNeedTypeAnnotation]
+if 1:
+    x = [] # E:5: Need type annotation for 'x' (hint: "x: List[<type>] = ...")
+[builtins fixtures/list.pyi]
+
+[case testColumnCallToUntypedFunction]
+# flags: --disallow-untyped-calls
+def f() -> None:
+    (g(1))  # E:6: Call to untyped function "g" in typed context
+
+def g(x):
+    pass
+
+[case testColumnInvalidArguments]
+def f(x, y): pass
+(f()) # E:2: Too few arguments for "f"
+(f(y=1)) # E:2: Missing positional argument "x" in call to "f"
+
+[case testColumnListOrDictItemHasIncompatibleType]
+from typing import List, Dict
+x: List[int] = [  # E:16: List item 0 has incompatible type "str"; expected "int"
+    'x']
```

Comment:
This may break some `# type: ignore`s, but I would set the context to the item expression, not the whole list expression.

Add a TODO item also about this?